lib/libc/sys/select.c

```diff
@@ -50,8 +43,11 @@ int select(int nfds, fd_set *restrict readfds, fd_set *restrict writefds,
         FD_ZERO(readfds);
     if (writefds != NULL)
         FD_ZERO(writefds);
-    
-    ret = kevent(kq, events, nevents, events, nevents, timeout == NULL ? NULL : &timeout_ts);
+
+    if (sigmask && sigprocmask(SIG_SETMASK, sigmask, &sigs))
+        return -1;
```

Comment:
`kq` is leaking here. The same for `events`.

Reply:
Since the `sigprocmask` has been moved before `kqueue` (https://github.com/cahirwpz/mimiker/pull/1419/commits/2dc7f82d58a0e638af3ca13518660648daae6eab), no more leaks occur.